simtools/model/telescope_model.py

```diff
@@ -574,7 +587,15 @@ def _loadCamera(self):
             cameraConfigFile=cameraConfigFilePath,
             focalLength=focalLength
         )
-        return
+
+    def _loadSimtelConfigWriter(self):
+        if not hasattr(self, 'simtelConfigWriter'):
+            self.simtelConfigWriter = SimtelConfigWriter(
+                site=self.site,
+                telescopeName=self.telescopeName,
+                modelVersion=self.modelVersion,
+                label=self.label
+            )
 
     def isASTRI(self):
```

Comment:
Why do we need isASTRI? I Think I've asked it before.

Reply:
We need this to fix an incompatibility problem between the model and sim_telarray (testeff in particular).

The testeff program, used by our CameraEfficiency, does not deal with 2D camera_filters and mirror_reflectivity. So we have to change these files to an 1D version of them (averaged).

Ideally, this problem should be fixed in testeff, and then we can make our code cleaner.

Reply:
OK, but both is handled by the optics module? And does the GCT camera will have a window where we want to use 2D efficiencies?

Reply:
This is handled by the camera_efficiency module, because it's an specific issue of testeff.
At the moment only ASTRI (old ASTRI design and the new unified SST design) are affected by that.

The problem is that we have to replace the 2D files by an specific file with the average. E.g. 'transmission_astri_window_average.dat'.

If this is not fixed in sim_telarray soon, we can spend some time on writing a tool to turn a 2D file into a 1D average file.
I'd assume that this is feasable.

Reply:
OK - but wouldn't it the be more appropriate to call the function not isAStri() but something like 'has2DEfficiencyTables()' or so? 

In a couple of years, noone will remember what ASTRI was, as we all speak of CTA SSTs...

Reply:
I already thought about this issue and I'm also not happy with the solution. It turns out it is a very specific issue. 

This is what is done in camera_efficiency:

        cameraFilterFile = self._telescopeModel.getParameter('camera_filter')
        if self._telescopeModel.isASTRI() and self._telescopeModel.isFile2D('camera_filter'):
            self._logger.warning(
                'Camera filter file is being replaced by transmission_astri_window_average.dat'
                ' because testeff does not support 2D camera filters.'
            )
            cameraFilterFile = 'transmission_astri_window_average.dat'

We replace the file by an ASTRI specific file, so in the end we can't solve the specificity of the code unless we solve the problem for real.

This means that if has2DEfficiencyTables() is True and it's not an ASTRI telescope, we still don't have a solution yet.

I'd prefer to keep it like this and open an issue requesting the real solution (either by in tesfeff or making the 2D files into 1D)

Reply:
OK. Could you please open an issue for this, so that we don't forget what is happening?

Reply:
Done!